python/pyarrow/_dataset.pyx

```diff
@@ -2239,10 +2233,6 @@ cdef class Scanner(_Weakrefable):
     use_threads : bool, default True
         If enabled, then maximum parallelism will be used determined by
         the number of available CPU cores.
-    use_async : bool, default False
```

Comment:
It seems that in Python, this keyword was not actually deprecated (in docs or deprecation warning)? 
Therefore, we could maybe still temporary just keep this keyword in the signature (without that it does anything, except from raising a warning if specified by the user, or something like that)

Reply:
I added the parameters back in.  PTAL.  One thing I wasn't sure of...

All of the places the parameter was declared it was a `bint` which means I couldn't change the default to `None` which means there was no good way to tell if the user specified it or not.

So I just made the default True and emit the warning if the user specifies `False`.  However, if a user has:

`dataset.to_table(use_async=True)` they will not get the warning.

Alternatively, I could change the cython type from bint to object.  Then I should be able to warn on both `use_async=True` and `use_async=False`.  Would this possible have any backwards compatibility ramifications?

Reply:
This was bugging me so I went ahead and played around with it.  I don't see any consequences if I change `bint use_async=True` to `object use_async = None` and that allows me to emit the deprecation warning on both `use_async=True` and `use_async=False` (i.e. emit a warning if the user uses the flag in any way).

Reply:
I also can't think of any concern for changing `bint` to `object` (the other way around would limit what you can pass, but for here it just broadens what you can pass, so that should not be a compatibility issue)